recipe-server/normandy/base/decorators.py

```diff
@@ -25,3 +27,23 @@ def wrapped_view(*args, **kwargs):
         with transaction.atomic(), revisions.create_revision(manage_manually=True):
             return view_func(*args, **kwargs)
     return wraps(view_func)(wrapped_view)
+
+
+def api_cache_control(**kwargs):
+    """
+    Adds cache headers to a view using our API cache header defaults.
+    """
+    if settings.API_CACHE_ENABLED:
+        directives = {
+            'public': True,
+            'max_age': settings.API_CACHE_TIME,
+        }
+    else:
+        directives = {
+            'no_cache': True,
+            'no_store': True,
+            'must_revalidate': True,
+        }
+
+    directives.update(kwargs)
```

Comment:
I don't think so, because it would still send `no-cache` and `no-store`, which (according to the [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)) forces the cache to submit to the origin server for revalidation.

I just tested this via compose, and with `DJANGO_API_CACHE_ENABLED=false` in the environment, all my requests to the action implementation view have `X-Cache: MISS` in the headers. Without it, they all have `X-Cache: HIT`.